### PR TITLE
render: Assert that texture dimensions are > 0

### PR DESCRIPTION
**Pull request by nyorain**
Assert that the case found in #2771 does not happen.

### CODE REVIEW

File: render/wlr_texture.c [hunk 1]
```diff
@@ -22,6 +22,8 @@ void wlr_texture_destroy(struct wlr_texture *texture) {
 struct wlr_texture *wlr_texture_from_pixels(struct wlr_renderer *renderer,
 		uint32_t fmt, uint32_t stride, uint32_t width, uint32_t height,
 		const void *data) {
+	assert(width > 0);
+	assert(height > 0);
```
Comment (emersion) [inline render/wlr_texture.c:26]: Maybe also assert that `stride > 0`?

Comment (nyorain) [reply render/wlr_texture.c:26]: `assert(data)` as well?

Comment (nyorain) [reply render/wlr_texture.c:26]: And `stride >= width` rather than `stride > 0`?

Comment (emersion) [reply render/wlr_texture.c:26]: Yeah, I guess that makes sense too

Comment (emersion) [reply render/wlr_texture.c:26]: I'm fine with `stride >= width` yeah, even if it won't catch all errors.

**Review by emersion — APPROVED** [review]
Thanks!

**Comment by ggreer** [comment]
I'm not sure why, but this caused Firefox to crash sway (but only if I restored session, not if I created a new profile). I removed the asserts and added logging: https://gist.github.com/ggreer/48540d062c94db18de0a0809fee9c1fa

```
00:00:09.645 [ERROR] [wlr] [render/wlr_texture.c:29] fmt: 875713089 stride: 0 width: 39886 height: 32
```

I then removed the `assert(data)` line and Firefox started up just fine. I must have closed the tab(s) responsible, because even after going back to wlroots master, I can no longer cause the crash.

File: render/wlr_texture.c [hunk 2]
```diff
@@ -22,6 +22,10 @@ void wlr_texture_destroy(struct wlr_texture *texture) {
...
+	assert(stride >= width);
```
Comment (emersion) [inline render/wlr_texture.c:27]: Now that I think of it, the stride can be smaller than the width for formats with a number of bits per pixel lower than 8. We don't support any of these yet.

**Comment by emersion** [comment]
Hm, a zero stride sounds like a client bug. But we shouldn't die on it. I'm surprised `wl_shm` doesn't catch it.

**Pull request merged**